zulipterminal/ui_tools/buttons.py

```diff
@@ -37,22 +37,25 @@ def __init__(self, controller: Any, caption: str,
         self.text_color = text_color
         self.show_function = show_function
         super().__init__("")
-        self._w = self.widget(count)
+        self.update_count(count)
         self.controller = controller
         urwid.connect_signal(self, 'click', self.activate)
 
     def update_count(self, count: int) -> None:
         self.count = count
-        self._w = self.widget(count)
-
-    def widget(self, count: int) -> Any:
-        if count < 0:
-            count_text = 'M'  # Muted
-        elif count == 0:
+        if count == 0:
             count_text = ''
         else:
             count_text = str(count)
+        self._w = self.widget(count_text)
+
+    def mark_button_muted(self) -> None:
+        # Ensure self.count has the correct unread counts
+        # by not updating it.
+        count_text = 'M'
+        self._w = self.widget(count_text)
```

Comment:
I wonder if it's worth introducing this method, since we already have `mark_muted` in the `StreamButton` (and removed it from `TopButton` as other buttons don't generally mute)? Perhaps override `update_count` instead, and call 'widget' where appropriate? (I'd be open to making that method more descriptive)

Reply:
Sure. I realized that a similar approach you suggested would be better for this.